app/scripts/lib/account-tracker.test.js

```diff
@@ -38,104 +38,98 @@ const mockAccounts = {
+    chainId: '0x1',
+  }).provider;
+
+  const blockTrackerFromHookStub = new EventEmitter();
+  blockTrackerFromHookStub.getCurrentBlock = noop;
+  blockTrackerFromHookStub.getLatestBlock = noop;
+
+  const getNetworkClientByIdStub = jest.fn().mockReturnValue({
+    configuration: {
       chainId: '0x1',
-    }).provider;
-
-    blockTrackerFromHookStub = new EventEmitter();
-    blockTrackerFromHookStub.getCurrentBlock = noop;
-    blockTrackerFromHookStub.getLatestBlock = noop;
-
-    getNetworkClientByIdStub = jest.fn().mockReturnValue({
-      configuration: {
-        chainId: '0x1',
-      },
-      blockTracker: blockTrackerFromHookStub,
-      provider: providerFromHook,
-    });
+    },
+    blockTracker: blockTrackerFromHookStub,
+    provider: providerFromHook,
+  });
 
-    getNetworkIdentifierStub = jest.fn();
-
-    controllerMessenger = new ControllerMessenger();
-    controllerMessenger.registerActionHandler(
-      'AccountsController:getSelectedAccount',
-      () => ({
-        id: 'accountId',
-        address: SELECTED_ADDRESS,
-      }),
-    );
-
-    accountTracker = new AccountTracker({
-      provider,
-      blockTracker: blockTrackerStub,
-      getNetworkClientById: getNetworkClientByIdStub,
-      getNetworkIdentifier: getNetworkIdentifierStub,
-      preferencesController: {
-        store: {
-          getState: () => ({
-            useMultiAccountBalanceChecker,
-          }),
-          subscribe: noop,
-        },
+  const controllerMessenger = new ControllerMessenger();
+  controllerMessenger.registerActionHandler(
+    'AccountsController:getSelectedAccount',
+    () => ({
+      id: 'accountId',
+      address: SELECTED_ADDRESS,
+    }),
+  );
+
+  return new AccountTracker({
+    provider,
+    blockTracker: blockTrackerStub,
+    getNetworkClientById: getNetworkClientByIdStub,
+    getNetworkIdentifier: jest.fn(),
+    preferencesController: {
+      store: {
+        getState: () => ({
+          useMultiAccountBalanceChecker,
+        }),
+        subscribe: noop,
       },
-      onboardingController: {
-        store: {
-          subscribe: noop,
-          getState: () => ({
-            completedOnboarding,
-          }),
-        },
+    },
+    onboardingController: {
+      store: {
+        subscribe: noop,
+        getState: () => ({
+          completedOnboarding,
+        }),
       },
-      controllerMessenger,
-      onAccountRemoved: (callback) => {
-        accountRemovedListener = callback;
-      },
-      getCurrentChainId: () => currentChainId,
-    });
+    },
+    controllerMessenger,
+    onAccountRemoved: noop,
+    getCurrentChainId: () => currentChainId,
+    ...accountTrackerOptions,
   });
+}
 
+describe('Account Tracker', () => {
   afterEach(() => {
     jest.resetAllMocks();
   });
 
   describe('start', () => {
     it('restarts the subscription to the block tracker and update accounts', async () => {
+      const blockTrackerStub = new EventEmitter();
```

Comment:
[nit] Also seems like this could be returned from the `buildAccountTracker` function since we stub it in there and want to listen to it in a number of these tests.

Reply:
addressed https://github.com/MetaMask/metamask-extension/pull/22233/commits/8bbd37575b900c94321bf9603a5bb1d3e23cb5a7